cmd/util/ledger/migrations/atree_register_migration.go

```diff
@@ -0,0 +1,508 @@
+				// to be sure, copy all cricket moments keys
+				newPayloads = append(newPayloads, value)
+				m.log.Info().Msgf("copying cricket moments key %s", id)
+				continue
+			}
+
+			m.rw.Write(migrationProblem{
+				Address: mr.Address.Hex(),
+				Key:     id.String(),
+				Size:    len(mr.Snapshot.Payloads),
+				Kind:    "not_migrated",
+				Msg:     fmt.Sprintf("%x", value),
+			})
+
+			size, err := payloadSize(key, value)
+			if err != nil {
+				return nil, fmt.Errorf("failed to get payload size: %w", err)
+			}
+
+			removedSize += size
+
+			// this is ok
+			// return nil, skippableAccountError
+		}
+	}
+
+	if statePayload == nil {
+		m.log.Error().Msg("state payload was not found")
+		return newPayloads, nil
+		//return nil, fmt.Errorf("state payload was not found")
+	}
+
+	// since some registers were removed, we need to update the storage used
+	if removedSize > 0 {
+		status, err := environment.AccountStatusFromBytes(statePayload.Value())
+		if err != nil {
+			return nil, fmt.Errorf("could not parse account status: %w", err)
+		}
+
+		status.SetStorageUsed(status.StorageUsed() - removedSize)
+
+		newPayload, err := newPayloadWithValue(statePayload, status.ToBytes())
+		if err != nil {
+			return nil, fmt.Errorf("cannot create new payload with value: %w", err)
+		}
+
+		statePayload = newPayload
+	}
+
+	newPayloads = append(newPayloads, statePayload)
+
+	return newPayloads, nil
+}
+
+func (m *AtreeRegisterMigrator) cloneValue(
+	mr *migratorRuntime,
+	domain string,
+	key interpreter.StorageMapKey,
+	value interpreter.Value,
+) (interpreter.Value, error) {
+
+	if isCricketMomentsShardedCollection(mr, value) {
+		m.log.Info().Msg("migrating CricketMomentsShardedCollection")
+		value, err := cloneCricketMomentsShardedCollection(
+			m.log,
+			m.nWorkers,
+			mr,
+			domain,
+			key,
+			value,
+		)
+
+		if err != nil {
+			m.log.Info().Err(err).Msg("failed to clone value")
+			return nil, err
+		}
+
+		m.log.Info().Msg("done migrating CricketMomentsShardedCollection")
+		return value, nil
+	}
+
+	err := capturePanic(func() {
+		// force the value to be read entirely
+		value = value.Clone(mr.Interpreter)
+	})
+	if err != nil {
+		return nil, err
+	}
+	return value, nil
+}
+
+// capturePanic captures panics and converts them to errors
+// this is needed for some cadence functions that panic on error
+func capturePanic(f func()) (err error) {
+	defer func() {
+		if r := recover(); r != nil {
+			var stack [100000]byte
+			n := runtime2.Stack(stack[:], false)
+			fmt.Printf("%s", stack[:n])
+
+			switch x := r.(type) {
+			case runtime.Error:
+				err = fmt.Errorf("runtime error @%s: %w", x.Location, x)
+			case error:
+				err = x
+			default:
+				err = fmt.Errorf("panic: %v", r)
+			}
+		}
+	}()
+	f()
+
+	return
+}
+
+// convert all domains
+var domains = []string{
+	common.PathDomainStorage.Identifier(),
+	common.PathDomainPrivate.Identifier(),
+	common.PathDomainPublic.Identifier(),
+	runtime.StorageDomainContract,
+}
```

Comment:
I think we need to add one more domain `inbox`, which also uses storage map under the hood.  

https://github.com/onflow/cadence/blob/24f277e481b23b683a11f75d487d685517c66c2f/runtime/stdlib/account.go#L911